src/main/java/com/redhat/cloud/notifications/routers/LivenessService.java

```diff
@@ -41,18 +38,10 @@
 //    }
 
     Uni<Boolean> postgresConnectionHealth() {
-        return connectionPublisherUni.toMulti()
-                .onItem().transformToMulti(c -> Multi.createFrom().resource(() -> c,
-                        c2 -> {
-                            Flux<PostgresqlResult> execute = c2.createStatement("SELECT COUNT(1)").execute();
-
-                            return execute.flatMap(postgresqlResult -> postgresqlResult.map((row, rowMetadata) -> true));
-                        })
-                        .withFinalizer(postgresqlConnection -> {
-                            postgresqlConnection.close().subscribe();
-                        }))
-                .merge().toUni()
-                .onFailure().recoverWithItem(false);
+        return session.createNativeQuery("SELECT 1")
```

Comment:
Fun fact: the Oracle driver used to not go to the DB to answer this, but did it locally, so that this query completely failed as connection test.

Reply:
Oh that's why there was a `COUNT` then. Thanks for the story, I didn't know this could happen.